pymc4/random_variables/continuous.py

```diff
@@ -42,6 +49,165 @@ def _base_dist(self, *args, **kwargs):
+            parameters=parameters,
+            graph_parents=[self._corr, self._skew],
+            name=name)
+    
+    @property
+    def corr(self):
+        """correlation matrix"""
+        return self._corr
+    
+    @property
+    def Omega(self):
+        shape = concat([self.batch_shape_tensor(), array_ops.shape(self.corr)[-2:]], 0)
+        return broadcast_to(self.corr, shape)
+    
+    # why doesn't TFP Distribution do this?
+    @property
+    def skew(self):
+        """skew vector"""
+        return self._skew
+    
+    @property
+    def alpha(self):
+        alpha = expand_dims(self.skew, -1) # makes a (col., n x 1) vector 
+        shape = concat([self.batch_shape_tensor(), array_ops.shape(alpha)[-2:]], 0)
+        return broadcast_to(alpha, shape)
+    
+    def _mean(self):
+        return (2./pi)**(.5) * self.get_delta()
+    
+    def _variance(self):
+        mu = self.mean()
+        return self.Omega - matmul(mu, mu, transpose_b=True)
+    
+    #@property # is property a good idea here, since there are computations involved?
+    #def delta(self):
+     #   return get_delta if not calculated
+    
+    def get_delta(self):
+        Omegaalpha = self.Omega @ self.alpha
+        return Omegaalpha @ (1 + matmul(self.alpha, Omegaalpha, transpose_a=True))**(-.5)
+    
+    #@property
+    def get_Omega_star(self):
+        """correlation matrix of underlying normal distribution"""
+        delta = self.get_delta()
+        Omega = self.Omega
+        one_shp = array_ops.shape(delta)
+        # one_shp[-2] = 1 doesn't work
+        # https://github.com/tensorflow/tensorflow/issues/14132
+        # so instead, ...
+        one_shp = concat([one_shp[:-2] , [1,1] ], 0)
+        one = ones(one_shp)
+        col0 = concat([one, delta], -2)
+        
+        delta_dim = tf_range(rank(delta))
+        delta_dim = concat([delta_dim[:-2], [delta_dim[-1]], [delta_dim[-2]]], 0)
+        col1 = concat([transpose(delta, perm=delta_dim ), Omega], -2)
+        return concat( [col0, col1], -1)
+        
+    def _batch_shape_tensor(self): 
+        # In other words, take the last 1 or 2 dims ( take as col. vector or matrix)
+        # and 'broadcast' the leading dims
+        return array_ops.broadcast_dynamic_shape(
+            array_ops.shape(self._corr)[:-2],
+            array_ops.shape(self._skew)[:-1]) # Tensor
+
+    def _batch_shape(self):
+        return array_ops.broadcast_static_shape(
+            self._corr.get_shape()[:-2],
+            self._skew.get_shape()[:-1])  # TensorShape
+    
+    def _event_shape_tensor(self):
+        return constant_op.constant([array_ops.shape(self.skew)[-1]], dtype=dtypes.int32)
+
+    def _event_shape(self):
+        return tensor_shape.TensorShape([self.skew.get_shape()[-1].value])
+    
+    def _sample_n(self, n, seed=None):
+        nd = self.event_shape[0] + 1
+        N = tfd.MultivariateNormalFullCovariance(loc=zeros(nd), covariance_matrix=self.get_Omega_star())
```

Comment:
for structured covariance matrices like the one you're using here it might be much more efficient (i.e. save the N^3 cholesky operation that MVNFull does by default) if you can use `tf.linalg.LinearOperator` and `tfd.MVNLinearOperator`